Language/Classes/Constructors/Factories/redirecting_to_itself_t04.dart

```diff
@@ -2,35 +2,39 @@
 // for details. All rights reserved. Use of this source code is governed by a
 // BSD-style license that can be found in the LICENSE file.
 
-/// @assertion It is a compile-time error if a redirecting factory constructor
-/// redirects to itself, either directly or indirectly via a sequence of
-/// redirections.
+/// @assertion A redirecting factory constructor q′ is redirection-reachable
+/// from a redirecting factory constructor q iff q′ is the redirectee
+/// constructor of q, or q′′ is the redirectee constructor of q and q′ is
+/// redirection-reachable from q′′. It is a compile-time error if a redirecting
+/// factory constructor is redirection-reachable from itself.
+///
 /// @description Checks that compile-error occurs when factory constructor
 /// redirects to itself through a cycle.
 /// @author ilya
 
-
 class F1 {
   factory F1() = F2;
 //               ^^
 // [analyzer] unspecified
 // [cfe] unspecified
 }
 
-class F2 {
+class F2 extends F1 {
   factory F2() = F3;
 //               ^^
 // [analyzer] unspecified
 // [cfe] unspecified
 }
 
-class F3 {
+class F3 extends F2 {
   factory F3() = F1;
```

Comment:
We still don't have a cycle of otherwise correct factory redirections, because `F1` isn't a subtype of `F3`. I don't think there is any other way to do this than using a single class (say, `F1`) and then having redirecting factory constructors in that class redirect to each other in a cycle.